rgw/admin/user.go

```diff
@@ -46,6 +48,12 @@ type UserKeySpec struct {
 	SecretKey string `json:"secret_key" url:"secret-key"`
 }
 
+type userStat struct {
```

Comment:
Is this type meant to be used outside go ceph (by consumers)? If so, shouldn't this be an exported type?

Reply:
Not really. It is not similar to `UserKeySpec`. The consumer should not add info, only the API will fill this one. If it make more sense I can define this structure internal to `User` only instead of `userStat`

Reply:
I don't think it matters if you're reading or writing to the API. You're defining a public attribute of a public type a private type. It strikes me as the wrong approach. CC @ansiwen to check me here.

Reply:
I think it's ok, it just avoids that someone outside creates a value of that sub-struct. As long as there is no exported function that returns that sub-type directly it should be fine. I assume no linter is complaining about this, right? (Not like the "returning unexported types in exported functions can be annoying" warning.) In general I like the unexported-by-default approach.

Reply:
@ansiwen : No, I didn't see any such warnings while building the code or running golint

Reply:
OK, so I guess it's not "wrong" from a low-level perspective. But I'm still going to insist for the reason of documentation. Putting a non-public type in a public struct, that I'm assuming users are expected to make use of to read the stats are not going to have any idea about what fields userStat has without reading the sources. 

Example: run `godo` in the go-ceph dir with this pr checked out. I see:
```
 User is GO representation of the json output of a user creation

type User struct {
    ID                  string        `json:"user_id" url:"uid"`
    DisplayName         string        `json:"display_name" url:"display-name"`
    Email               string        `json:"email" url:"email"`
    Suspended           *int          `json:"suspended" url:"suspended"`
    MaxBuckets          *int          `json:"max_buckets" url:"max-buckets"`
    Subusers            []interface{} `json:"subusers"`
    Keys                []UserKeySpec `json:"keys"`
    SwiftKeys           []interface{} `json:"swift_keys"`
    Caps                []UserCapSpec `json:"caps"`
    OpMask              string        `json:"op_mask"`
    DefaultPlacement    string        `json:"default_placement"`
    DefaultStorageClass string        `json:"default_storage_class"`
    PlacementTags       []interface{} `json:"placement_tags"`
    BucketQuota         QuotaSpec     `json:"bucket_quota"`
    UserQuota           QuotaSpec     `json:"user_quota"`
    TempURLKeys         []interface{} `json:"temp_url_keys"`
    Type                string        `json:"type"`
    MfaIds              []interface{} `json:"mfa_ids"`
    KeyType             string        `url:"key-type"`
    Tenant              string        `url:"tenant"`
    GenerateKey         *bool         `url:"generate-key"`
    PurgeData           *int          `url:"purge-data"`
    GenerateStat        *bool         `url:"stats"`
    Stat                userStat      `json:"stats"`
}

```

But I see nothing in the docs about what userStat is or how I can make use of it. We really ought to make it useful and make userStat public. If you don't want to make it public we need methods to read the stats, but I really don't think that's better. Plus, if you do that you need to add a doc comment to the `Stat` field pointing readers at these accessor methods so people aren't left wondering what this thing's for.